dm/master/shardddl/optimist.go

```diff
@@ -312,26 +320,32 @@ func (o *Optimist) buildLockJoinedAndTTS(ifm map[string]map[string]map[string]ma
 // recoverLocks recovers shard DDL locks based on shard DDL info and shard DDL lock operation.
 func (o *Optimist) recoverLocks(
 	ifm map[string]map[string]map[string]map[string]optimism.Info,
-	opm map[string]map[string]map[string]map[string]optimism.Operation) error {
+	opm map[string]map[string]map[string]map[string]optimism.Operation,
+	colm map[string]map[string]map[string]map[string]map[string]struct{}) error {
+	o.lk.SetColumnMap(colm)
+	defer o.lk.SetColumnMap(nil)
 	// construct joined table based on the shard DDL info.
 	o.logger.Info("build lock joined and tts")
 	lockJoined, lockTTS := o.buildLockJoinedAndTTS(ifm)
 	// build lock and restore table info
 	o.logger.Info("rebuild locks and tables")
-	o.lk.RebuildLocksAndTables(ifm, lockJoined, lockTTS)
+	o.lk.RebuildLocksAndTables(o.cli, ifm, lockJoined, lockTTS)
```

Comment:
I mean remove `SetColumnMap`. Instead, iterate through `lk.locks` and `colm` and set them